src/main/java/org/openrewrite/java/migrate/apache/commons/lang/IsNotEmptyToJdk.java

```diff
@@ -0,0 +1,90 @@
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.apache.commons.lang;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+public class IsNotEmptyToJdk extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Replace any StringUtils#isEmpty(String) and #isNotEmpty(String)";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace any `StringUtils#isEmpty(String)` and `#isNotEmpty(String)` with `s == null || s.isEmpty()` and `s != null && !s.isEmpty()`.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaVisitor<ExecutionContext>() {
+
+            private final MethodMatcher commonsIsEmptyMatcher = new MethodMatcher(org.apache.commons.lang3.StringUtils.class.getName() + " isEmpty(..)");
+            private final MethodMatcher commonsIsNotEmptyMatcher = new MethodMatcher(org.apache.commons.lang3.StringUtils.class.getName() + " isNotEmpty(..)");
+            private final MethodMatcher plexusIsEmptyMatcher = new MethodMatcher(org.codehaus.plexus.util.StringUtils.class.getName() + " isEmpty(..)");
+            private final MethodMatcher plexusIsNotEmptyMatcher = new MethodMatcher(org.codehaus.plexus.util.StringUtils.class.getName() + " isNotEmpty(..)");
+            private final MethodMatcher mavenSharedIsEmptyMatcher = new MethodMatcher(org.apache.maven.shared.utils.StringUtils.class.getName() + " isEmpty(..)");
+            private final MethodMatcher mavenSharedIsNotEmptyMatcher = new MethodMatcher(org.apache.maven.shared.utils.StringUtils.class.getName() + " isNotEmpty(..)");
+
+            private final JavaTemplate commonsIsEmptyTemplate = JavaTemplate.compile(this, "CommonsIsEmpty", (String s) -> (s == null || s.isEmpty())).build();
+            private final JavaTemplate commonsIsNotEmptyTemplate = JavaTemplate.compile(this, "CommonsIsNotEmpty", (String s) -> (s != null && !s.isEmpty())).build();
+            private final JavaTemplate plexusIsEmptyTemplate = JavaTemplate.compile(this, "PlexusIsEmpty", (String s) -> (s == null || s.isEmpty())).build();
+            private final JavaTemplate plexusIsNotEmptyTemplate = JavaTemplate.compile(this, "PlexusIsNotEmpty", (String s) -> (s != null && !s.isEmpty())).build();
+            private final JavaTemplate mavenSharedIsEmptyTemplate = JavaTemplate.compile(this, "MavenSharedIsEmpty", (String s) -> (s == null || s.isEmpty())).build();
+            private final JavaTemplate mavenSharedIsNotEmptyTemplate = JavaTemplate.compile(this, "MavenSharedIsNotEmpty", (String s) -> (s != null && !s.isEmpty())).build();
```

Comment:
As also indicated below: the `name` can only be a literal; otherwise we could have reduced the duplication here.

Reply:
As the templates look identical I am unsure why we can't just use the same two templates everywhere.

Reply:
You're right; had the same insight the moment I stepped away from the laptop; should do that more often! ;)